src/shared/regions/defaultRegionProvider.ts

```diff
@@ -4,98 +4,59 @@
-        if (this._areRegionsLoaded) {
-            return this._loadedRegions
-        }
-
-        let availableRegions: RegionInfo[] = []
-        try {
-            this.logger.info('Retrieving AWS endpoint data')
-
-            const endpointsContents = await this._resourceFetcher.get()
-
-            if (!endpointsContents) {
-                throw new Error('No endpoints data found')
-            }
-
-            const allEndpoints = JSON.parse(endpointsContents) as RawEndpoints
+        return this._loadedRegions
+    }
 
-            availableRegions = getRegionsFromEndpoints(allEndpoints)
+    private loadFromEndpointsProvider(provider: EndpointsProvider) {
+        const endpoints = provider.getEndpoints()
 
-            this._areRegionsLoaded = true
-            this._loadedRegions = availableRegions
-        } catch (err) {
-            this._areRegionsLoaded = false
-            this.logger.error('Unable to retrieve AWS endpoints: ', err as Error)
-            // TODO: now what, oneline + local failed...?
-            availableRegions = []
-            this._loadedRegions = []
+        if (endpoints) {
+            this.loadFromEndpoints(endpoints)
         }
+    }
+
+    private loadFromEndpoints(endpoints: Endpoints) {
+        // TODO : Support other Partition regions : https://github.com/aws/aws-toolkit-vscode/issues/188
+        this._loadedRegions = getRegionInfo(endpoints, 'aws')
 
-        return availableRegions
+        this.onRegionProviderUpdatedEmitter.fire()
     }
 }
 
-export function makeEndpointsResourceFetcher(extensionContext: vscode.ExtensionContext): ResourceFetcher {
-    return new CompositeResourceFetcher(
-        new HttpResourceFetcher(endpointsFileUrl),
-        new FileResourceFetcher(ext.manifestPaths.endpoints)
-    )
-}
+function getRegionInfo(endpoints: Endpoints, partitionId: string): RegionInfo[] {
+    let regions: RegionInfo[] = []
 
-export function getRegionsFromPartition(partition: RawPartition): RegionInfo[] {
-    return Object.keys(partition.regions).map(
-        regionKey => new RegionInfo(regionKey, `${partition.regions[regionKey].description}`)
-    )
-}
+    endpoints.partitions
+        .filter(partition => partition.id === partitionId)
+        .forEach(partition => {
+            regions = regions.concat(partition.regions.map(asRegionInfo))
+        })
 
-export function getRegionsFromEndpoints(endpoints: RawEndpoints): RegionInfo[] {
-    return (
-        endpoints.partitions
-            // TODO : Support other Partition regions : https://github.com/aws/aws-toolkit-vscode/issues/188
-            .filter(partition => partition.partition && partition.partition === 'aws')
-            .reduce((accumulator: RegionInfo[], partition: RawPartition) => {
-                accumulator.push(...getRegionsFromPartition(partition))
+    return regions
+}
 
-                return accumulator
-            }, [])
-    )
+function asRegionInfo(region: Region): RegionInfo {
```

Comment:
Why are these two separate types? what does the separation do?

Reply:
I would like to phase out RegionInfo and switch to `Region` (in endpoints.ts). I would like to do that in a subsequent PR, leaving this one solely to introduce the endpoint-region provider separation.